imap_processing/hi/hi_l2.py

```diff
@@ -98,34 +101,64 @@ def generate_hi_map(
         The sky map with all the PSET data projected into the map.
     """
     output_map = descriptor.to_empty_map()
+    vars_to_bin = ["counts", "exposure_factor", "bg_rates", "bg_rates_unc", "obs_date"]
+
+    if descriptor.frame_descriptor == "hf":
+        vars_to_bin.append("energy_sc")
 
     if not isinstance(output_map, RectangularSkyMap):
         raise NotImplementedError("Healpix map output not supported for Hi")
 
-    # TODO: Implement Compton-Getting correction
-    if descriptor.frame_descriptor != "sf":
-        raise NotImplementedError("CG correction not implemented for Hi")
+    cached_esa_steps = None
 
     for pset_path in psets:
         logger.info(f"Processing {pset_path}")
-        pset = HiPointingSet(pset_path, spin_phase=descriptor.spin_phase)
-
-        # Background rate and uncertainty are exposure time weighted means in
-        # the map.
+        pset = HiPointingSet(pset_path)
+
+        # Store the first PSET esa_energy_step values and make sure every PSET
+        # contains the same set of esa_energy_step values.
```

Comment:
Is this a hard requirement? I'm wondering if you can just recompute what you need to if the cache changes.

```
if not np.array_equal(cached_esa_steps, pset.data["esa_energy_step"].values):
    # recompute cache
```